src/db/tables/onerep_scans.ts

```diff
@@ -13,26 +13,48 @@ import { OnerepScanResultRow, OnerepScanRow } from "knex/types/tables";
 
 const knex = initKnex(knexConfig);
 export interface LatestOnerepScanData {
-  scan: OnerepScanRow | null;
   results: OnerepScanResultRow[];
 }
 
+async function getAllScansForProfile(
+  onerepProfileId: number,
+): Promise<OnerepScanRow[]> {
+  const scans = await knex("onerep_scans")
+    .where("onerep_profile_id", onerepProfileId)
+    .orderBy("created_at", "desc");
+
+  return scans;
+}
+
+async function getScanResults(
+  onerepScanId: number,
+): Promise<OnerepScanResultRow[]> {
+  const scanResults = await knex("onerep_scan_results").where(
+    "onerep_scan_id",
+    onerepScanId,
+  );
+
+  return scanResults;
+}
+
 async function getLatestOnerepScanResults(
   onerepProfileId: number,
 ): Promise<LatestOnerepScanData> {
-  const scan = await knex("onerep_scans")
-    .first()
+  const result = await knex("onerep_scan_results")
+    .select("onerep_scan_id")
+    .from("onerep_scans")
     .where("onerep_profile_id", onerepProfileId)
     .orderBy("created_at", "desc");
 
+  const scanIds = result.map((scan_id) => scan_id["onerep_scan_id"]);
+
   const results =
-    typeof scan === "undefined"
+    typeof scanIds === "undefined"
```

Comment:
I just simplified this ... yes it could be done with a join instead, I'm not sure it would be much more efficient (it wouldn't require reading into the app but we wouldn't expect the result set of scan IDs to be very large). Whether it's more readable is a separate question :) I'll try doing it with a `join`.